arroyo/backends/kafka/consumer.py

```diff
@@ -246,40 +252,63 @@ def assignment_callback(
         ) -> None:
             self.__state = KafkaConsumerState.ASSIGNING
 
-            try:
-                assignment: MutableSequence[ConfluentTopicPartition] = []
+            if self.__incremental_cooperative is True:
```

Comment:
This diff got pretty weird because of indentation. This block from here to line 273 is new though, and could use a close review. It's quite different to what happens with eager rebalancing (lines 276-305), which is essentially unchanged.